### PR TITLE
[IMP] *: add view info in session info

### PR DESCRIPTION
**Pull request by Polymorphe57**
We have noted that some view information like "icon" that was sometimes modified in some js classes were in fact not variable for a given (server side) type of view. Here we set a key "view_info" in `__session_info__` with the view information and make the action service use it.
This will allow us to easily lazy load views.

In preparation of Task `3546321`

### CODE REVIEW

File: addons/web/static/src/@types/registries/views_registry.d.ts [hunk 1]
```diff
@@ -27,16 +27,11 @@ declare module "registries" {
...
-        display_name: string;
-        icon: string;
-        isMobileFriendly?: boolean;
         Model: typeof Model;
-        multiRecord: boolean;
         props(genericProps: ViewInfo, viewDescr: ViewsRegistryItemShape, config: object): object;
         Renderer: typeof Component;
         searchMenuTypes?: ("filter" | "groupBy" | "comparison" | "favorite")[];
         SearchModel?: typeof SearchModel;
-        type: string;
```
Comment (Polymorphe57) [inline addons/web/static/src/@types/registries/views_registry.d.ts:39]: not sure to remove type

File: addons/web/static/tests/_framework/mock_server_state.hoot.js [hunk 2]
```diff
@@ -29,6 +29,64 @@ const DEFAULT_LUXON_SETTINGS = {
     defaultZone: Settings.defaultZone,
     defaultWeekSettings: Settings.defaultWeekSettings,
 };
+
+const VIEW_INFO = {
```
Comment (Polymorphe57) [inline addons/web/static/tests/_framework/mock_server_state.hoot.js:33]: how to set that correctly in test? receive it from server?

File: addons/web/static/src/webclient/actions/action_service.js [hunk 3]
```diff
@@ -1103,8 +1103,13 @@ export function makeActionManager(env, router = _router) {
     async function _executeActWindowAction(action, options) {
         const views = [];
         for (const [, type] of action.views) {
-            if (type !== "search" && viewRegistry.contains(type)) {
-                views.push(viewRegistry.get(type));
+            if (type !== "search" && session.view_info[type]) {
+                const {
+                    icon,
+                    display_name,
```
Comment (Polymorphe57) [inline addons/web/static/src/webclient/actions/action_service.js:1109]: construct it with
```js
const f = (str) => str[0].toUpperCase() + str.slice(1)
```
?
QWeb is not used here

File: odoo/addons/base/models/ir_ui_view.py [hunk 4]
```diff
@@ -141,6 +141,43 @@ def get_view_arch_from_file(filepath, xmlid):
...
+    },
+    'gantt': {
+        'icon': 'fa fa-tasks',
+        'display_name': 'Gantt',
+    },
+    'kanban': {
+        'icon': 'oi oi-view-kanban',
+        'display_name': 'Kanban',
+    },
+    'search': {
```
Comment (Polymorphe57) [inline odoo/addons/base/models/ir_ui_view.py:173]: remove?

File: odoo/addons/base/models/ir_ui_view.py [hunk 5]
```diff
@@ -141,6 +141,43 @@ def get_view_arch_from_file(filepath, xmlid):
...
+        'display_name': 'Search',
+    },
+    'qweb': {
+        'display_name': 'QWeb',
```
Comment (Polymorphe57) [inline odoo/addons/base/models/ir_ui_view.py:177]: remove?

File: odoo/addons/base/models/ir_ui_view.py [hunk 6]
```diff
@@ -563,6 +572,18 @@ def default_view(self, model, view_type):
...
+    def get_view_info(self):
+        view_info = {}
+        for (type, display_name) in self.env['ir.model.fields'].get_field_selection('ir.ui.view', 'type'):
+            t = 'list' if type == 'tree' else type
+            view_info[t] = {
+                'display_name': display_name,
+                'icon': VIEW_ICONS.get(type, 'fa fa-question'),
+                'multi_record': False if type == 'form' else True,
+            }
+        return view_info
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply odoo/addons/base/models/ir_ui_view.py]: done

File: addons/mail/models/ir_ui_view.py [hunk 7]
```diff
@@ -1,5 +1,8 @@
 # -*- coding: utf-8 -*-
 from odoo import fields, models
+from odoo.addons.base.models.ir_ui_view import VIEW_ICONS
+
+VIEW_ICONS['activity'] =  'fa fa-clock-o'
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/mail/models/ir_ui_view.py]: > I also see that there is a dedicated module for `gantt`, `'gantt': 'fa fa-tasks'` should be moved to a new ir.ui_view override inside that module. I haven't verified the other entries.

Yes, I know but there is a bunch of things in the community addon mrp that are related to gantt
-> not possible right now

[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/mail/models/ir_ui_view.py]: I mean the type 'gantt' is added in the selection here

File: addons/web/models/ir_ui_view.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            if type_ != 'qweb'
+        }
+
+    def _get_view_info(self):
+        return {
+            'tree': {'icon': 'oi oi-view-list'},
+            'form': {'icon': 'fa fa-address-card', 'multi_record': False},
+            'graph': {'icon': 'fa fa-area-chart'},
+            'pivot': {'icon': 'oi oi-view-pivot'},
+            'calendar': {'icon': 'fa fa-calendar'},
```
Comment (Julien00859) [inline addons/web/models/ir_ui_view.py:27]: not a blocker, feel free to ignore
shouldn't calendar be defined in the calendar module?

**Review by Julien00859 — APPROVED** [review]
python code

File: addons/web/models/ir_ui_view.py (hunk 8 above)
Comment (Polymorphe57) [reply addons/web/models/ir_ui_view.py:27]: maybe. Things should be clarified for this and gantt

**Review by aab-odoo — COMMENTED** [review]
nice

File: addons/web/static/src/webclient/actions/action_service.js [hunk 9]
```diff
@@ -1103,8 +1103,9 @@ export function makeActionManager(env, router = _router) {
     async function _executeActWindowAction(action, options) {
         const views = [];
         for (const [, type] of action.views) {
-            if (type !== "search" && viewRegistry.contains(type)) {
-                views.push(viewRegistry.get(type));
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/web/static/src/webclient/actions/action_service.js:1107]: I'll do that elsewhere

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**